.github/workflows/build-and-test.yml

```diff
@@ -36,12 +32,26 @@ jobs:
       run: dotnet build --no-restore --configuration Release
 
     - name: Test
-      run: dotnet test --no-build --no-restore --configuration Release --logger "trx;LogFileName=TestResults.trx"
+      run: dotnet test --no-build --no-restore --configuration Release --logger "junit;LogFilePath=test-result.xml"
 
-    # as per https://github.com/dorny/test-reporter?tab=readme-ov-file#recommended-setup-for-public-repositories
-    - name: Test Report
+    - name: Publish
```

Comment:
Why do we need to publish & upload artifacts on build & test?

Reply:
so you have an artifact that can be deployed